enforcer-rules/README.md

```diff
@@ -0,0 +1,94 @@
+# Linkage Checker Enforcer Rule
+
+This Maven enforcer rule verifies that the transitive dependency tree of `pom.xml` does not have
+[linkage error](../library-best-practices/glossary.md#types-of-conflicts-and-compatibility).
+
+## Class path and BOM flag
+
+The rule takes `bom` flag to control the behavior (by default `false`).
+
+- When `bom=false`, the rule checks the class path calculated from the project's `dependencies`
+  section and their transitive dependencies.
+- When `bom=true`, the rule checks a class path consisting of artifacts in the BOM's
+  `dependencyManagement` section and their transitive dependencies.
+
+# Usage
+
+Add the following plugin configuration to your `pom.xml`:
+
+```xml
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-enforcer-plugin</artifactId>
+        <version>3.0.0-M2</version>
+        <dependencies>
+          <dependency>
+            <groupId>com.google.cloud.tools.opensource</groupId>
+            <artifactId>linkage-checker-enforcer-rules</artifactId>
+            <version>0.1-SNAPSHOT</version>
+          </dependency>
+        </dependencies>
+        <executions>
+          <execution>
+            <id>enforce</id>
+            <goals>
+              <goal>enforce</goal>
+            </goals>
+            <configuration>
+              <rules>
+                <banLinkageErrors
+                    implementation="com.google.cloud.tools.opensource.enforcer.LinkageCheckerRule"/>
+              </rules>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+   ...
+```
+
+For a BOM project, set `bom` flag to true.
+
+```xml
+  <banLinkageErrors
+      implementation="com.google.cloud.tools.opensource.enforcer.LinkageCheckerRule">
+      <bom>true</bom>
+  </banLinkageErrors>
+```
+
+When you do not want to fail the rule, run Maven with `-Denforcer.fail=false`:
```

Comment:
This is all enforcer rules, not just this one, right? Probably don't need to mention this. The section on setting level to warn is enough.